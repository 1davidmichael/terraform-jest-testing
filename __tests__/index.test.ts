import * as hcl from "@cdktf/hcl2json";

describe("Terraform Snapshot", () => {
    it("Should have resources with required properties", async () => {
        const data = await hcl.convertFiles(".");
        expect(data).toMatchSnapshot();
    });
});
