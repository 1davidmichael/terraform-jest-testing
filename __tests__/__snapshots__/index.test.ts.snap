// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Terraform Snapshot Should have resources with required properties 1`] = `
{
  "resource": {
    "aws_s3_bucket": {
      "test_bucket": [
        {},
      ],
    },
    "aws_s3_bucket_public_access_block": {
      "block_public_access": [
        {
          "block_public_acls": true,
          "block_public_policy": true,
          "bucket": "\${aws_s3_bucket.test_bucket.id}",
          "ignore_public_acls": true,
          "restrict_public_buckets": true,
        },
      ],
    },
    "aws_s3_bucket_server_side_encryption_configuration": {
      "bucket_encryption": [
        {
          "bucket": "\${aws_s3_bucket.test_bucket.id}",
          "rule": [
            {
              "apply_server_side_encryption_by_default": [
                {
                  "sse_algorithm": "AES256",
                },
              ],
            },
          ],
        },
      ],
    },
    "aws_s3_bucket_versioning": {
      "versioning": [
        {
          "bucket": "\${aws_s3_bucket.test_bucket.id}",
          "versioning": [
            {
              "enabled": true,
            },
          ],
        },
      ],
    },
    "null_resource": {
      "resource1": [
        {},
      ],
      "resource2": [
        {},
      ],
      "resource3": [
        {},
      ],
    },
  },
}
`;
